Server/tests/test_hardening_security.py

```
@@ -0,0 +1,164 @@
+# --------------------------------------------------------------------------- #
+class TestBridgeTokenResolver:
+    def _fresh_home(self, monkeypatch, tmp_path):
+        monkeypatch.delenv("UNITY_MCP_BRIDGE_TOKEN", raising=False)
+        monkeypatch.setenv("HOME", str(tmp_path))
+        monkeypatch.setattr(Path, "home", classmethod(lambda cls: tmp_path))
+
+    def test_env_token_wins(self, monkeypatch, tmp_path):
+        from transport.legacy.unity_connection import resolve_bridge_token
+        self._fresh_home(monkeypatch, tmp_path)
+        # Even with a file present, the env var takes precedence.
+        d = tmp_path / ".unity-mcp"; d.mkdir()
```

Ruff (0.15.17)

[error] Multiple statements on one line (semicolon)

(E702)